src/utils/extentUtils.js

```diff
@@ -84,26 +84,33 @@ export function getExtentIntersectionWithCurrentProjection(
     ) {
         return null
     }
-    let extentInCurrentProjection = flattenExtent(extent)
+    let currentProjectionAsExtentProjection = currentProjection.bounds.flatten
+    let finalExtent = flattenExtent(extent)
```

Comment:
why declare finalExtent here and not just declare it later with 
`
let finalExtent = getExtentIntersection(flattenExtent(extent), currentProjectionAsExtentProjection)
`

Reply:
good point, now that it's not used in the next if block I could move the declaration further down